llama-index-core/tests/workflow/test_event.py

```diff
@@ -87,3 +88,18 @@ def test_event_dict_api():
     assert v == "bar"
     assert next(iter(ev)) == "a_new_key"
     assert ev.dict() == {"a_new_key": "bar"}
+
+
+def test_event_serialization():
+    ev = _TestEvent(param="foo", not_a_field="bar")
+    serializer = JsonSerializer()
+    serialized_ev = serializer.serialize(ev)
+    deseriazlied_ev = serializer.deserialize(serialized_ev)
+
+    assert type(deseriazlied_ev).__name__ == type(ev).__name__
+    deseriazlied_ev = cast(
+        _TestEvent,
+        deseriazlied_ev,
+    )
+    assert ev.param == deseriazlied_ev.param
+    assert ev._data == deseriazlied_ev._data
```

Comment:
This last assertion fails without the custom additions to Event.model_dump() in this PR.